File: src/main/java/com/airsaid/localization/translate/impl/openai/OpenAIResponse.java
/*
 * Copyright 2021 Airsaid. https://github.com/airsaid
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package com.airsaid.localization.translate.impl.openai;

import java.util.List;

public class OpenAIResponse {
    private List<Choice> choices;
    private Integer created;
    private String id;
    private String object;
    private Usage usage;

    public OpenAIResponse(List<Choice> choices, Integer created, String id, String object, Usage usage) {
        this.choices = choices;
        this.created = created;
        this.id = id;
        this.object = object;
        this.usage = usage;
    }

    public List<Choice> getChoices() {
        return choices;
    }

    public void setChoices(List<Choice> choices) {
        this.choices = choices;
    }

    public Integer getCreated() {
        return created;
    }

    public void setCreated(Integer created) {
        this.created = created;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getObject() {
        return object;
    }

    public void setObject(String object) {
        this.object = object;
    }

    public Usage getUsage() {
        return usage;
    }

    public void setUsage(Usage usage) {
        this.usage = usage;
    }

    public String getTranslation() {
        if (choices != null && !choices.isEmpty()) {
            String result = choices.get(0).getMessage().getContent();
            return result.trim();

        } else {
            return "";
        }
    }

    public static class Choice {
        private String finish_reason;
        private Integer index;
        private Message message;

        public Choice(String finish_reason, Integer index, Message message) {
            this.finish_reason = finish_reason;
            this.index = index;
            this.message = message;
        }

        public String getFinish_reason() {
            return finish_reason;
        }

        public void setFinish_reason(String finish_reason) {
            this.finish_reason = finish_reason;
        }

        public Integer getIndex() {
            return index;
        }

        public void setIndex(Integer index) {
            this.index = index;
        }

        public Message getMessage() {
            return message;
        }

        public void setMessage(Message message) {
            this.message = message;
        }
    }

    public static class Message {
        private String content;
        private String role;

        public Message(String content, String role) {
            this.content = content;
            this.role = role;
        }

        public String getContent() {
            return content;
        }

        public void setContent(String content) {
            this.content = content;
        }

        public String getRole() {
            return role;
        }

        public void setRole(String role) {
            this.role = role;
        }
    }

    public static class Usage {
        private Integer completion_tokens;
        private Integer prompt_tokens;
        private Integer total_tokens;

        public Usage(Integer completion_tokens, Integer prompt_tokens, Integer total_tokens) {
            this.completion_tokens = completion_tokens;
            this.prompt_tokens = prompt_tokens;
            this.total_tokens = total_tokens;
        }

        public Integer getCompletion_tokens() {
            return completion_tokens;
        }

        public void setCompletion_tokens(Integer completion_tokens) {
            this.completion_tokens = completion_tokens;
        }

        public Integer getPrompt_tokens() {
            return prompt_tokens;
        }

        public void setPrompt_tokens(Integer prompt_tokens) {
            this.prompt_tokens = prompt_tokens;
        }

        public Integer getTotal_tokens() {
            return total_tokens;
        }

        public void setTotal_tokens(Integer total_tokens) {
            this.total_tokens = total_tokens;
        }
    }
}

File: src/main/kotlin/com/airsaid/localization/translate/impl/openai/OpenAIResponse.kt
/*
 * Copyright 2021 Airsaid. https://github.com/airsaid
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package com.airsaid.localization.translate.impl.openai

data class OpenAIResponse(
    var choices: List<Choice>?,
    var created: Int?,
    var id: String?,
    var `object`: String?,
    var usage: Usage?
) {
    val translation: String
        get() {
            return if (!choices.isNullOrEmpty()) {
                val result = choices!![0].message?.content
                result?.trim() ?: ""
            } else {
                ""
            }
        }

    data class Choice(
        var finish_reason: String?,
        var index: Int?,
        var message: Message?
    )

    data class Message(
        var content: String?,
        var role: String?
    )

    data class Usage(
        var completion_tokens: Int?,
        var prompt_tokens: Int?,
        var total_tokens: Int?
    )
}